run_bot.py

```
@@ -65,13 +65,38 @@ def main():
+        sys.exit(2)
+
     # If running in cache-only mode, we skip authentication and proceed using only cached files.
     session = None
     if args.cache_mode != "cache-only":
         try:
             # Authenticate using the synchronous requests library
             session = gartan_login_and_get_session()
         except AuthenticationError as e:
+            if "Invalid username or password" in str(e):
+                logger.critical(
+                    f"🔒 CRITICAL: Authentication failed due to invalid credentials: {str(e)}"
```

Ruff (0.15.20)

[warning] Logging statement uses f-string

(G004)